### PR TITLE
Add VNC enabled templates

### PR DESCRIPTION
**Pull request by gavin-k-lee**

### CODE REVIEW

File: desktop/README.md [hunk 1]
```diff
@@ -0,0 +1,3 @@
+# RENKU desktop
+
+This project template showcases the latest VNC (virtual desktop) that RENKU comes with.
```
Comment (rokroskar) [inline desktop/README.md:3]: This is going to be the README of the actual project - probably want to add some different information here?

File: matlab/.gitignore [hunk 2]
```diff
@@ -0,0 +1,338 @@
...
+### VisualStudioCode Patch ###
+# Ignore all local history of files
+.history
+
+# End of https://www.gitignore.io/api/macos,python,R,linux,vim,emacs,visualstudiocode,intellij
+
+# Renku
+.renku.lock
+.renku/tmp
+.renku/cache
```
Comment (rokroskar) [inline matlab/.gitignore:338]: what about temporary files that matlab might produce?

File: matlab/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+# Copy MATLAB install from supplied Docker image
+COPY --from=matlab-install-stage ${BASE_ML_INSTALL_LOC} /usr/local/MATLAB
+
+# Put MATLAB on the PATH
+RUN ln -s /usr/local/MATLAB/bin/matlab /usr/local/bin/matlab
+
+# Install MATLAB dependencies
+# Reference: https://github.com/mathworks-ref-arch/container-images/tree/master/matlab-deps
+# The following are the dependencies for R2021a
+RUN export DEBIAN_FRONTEND=noninteractive && apt-get update && apt-get install --no-install-recommends -y \
```
Comment (rokroskar) [inline matlab/Dockerfile:45]: it's a lot of stuff here - should we make a matlab base image instead? If it's all in the template we don't actually have a good way of knowing if it even builds.

File: matlab/README.md [hunk 4]
```diff
@@ -0,0 +1,9 @@
+# {{ name }}
+
+{% if __project_description__ %}
+{{ __project_description__ }}
+{% endif %}
```
Comment (rokroskar) [inline matlab/README.md:5]: this is missing from the desktop readme above - probably want them to be consistent?

File: matlab/requirements.txt [hunk 5]
```diff
@@ -0,0 +1,2 @@
+jupyter-matlab-proxy
+numpy
```
Comment (rokroskar) [inline matlab/requirements.txt:2]: why do you need numpy for matlab? in any case, these should be a part of the base image, see above.

File: matlab/.renku/renku.ini [hunk 6]
```diff
@@ -0,0 +1,10 @@
+[interactive]
+default_url = /lab
```
Comment (rokroskar) [inline matlab/.renku/renku.ini:2]: not `/vnc`?

File: desktop/.renku/renku.ini [hunk 7]
```diff
@@ -0,0 +1,9 @@
+[interactive]
+default_url = /lab
```
Comment (rokroskar) [inline desktop/.renku/renku.ini:2]: `/vnc`?

File: desktop/.renku/renku.ini [hunk 8]
```diff
@@ -0,0 +1,9 @@
...
+cpu_request = 2
+mem_request = 8G
```
Comment (rokroskar) [inline desktop/.renku/renku.ini:4]: are these generic requirements?

File: desktop/.renku/renku.ini [hunk 9]
```diff
@@ -0,0 +1,9 @@
...
+
+[renku]
+autocommit_lfs = false
```
Comment (rokroskar) [inline desktop/.renku/renku.ini:7]: there be dragons...

File: matlab/.renku/renku.ini [hunk 10]
```diff
@@ -0,0 +1,10 @@
...
+cpu_request = 2
+mem_request = 8G
+disk_request = 4G
+
+[renku]
+autocommit_lfs = false
```
Comment (rokroskar) [inline matlab/.renku/renku.ini:8]: not sure this is on purpose?

File: qgis/Dockerfile [hunk 11]
```diff
@@ -0,0 +1,71 @@
+# For finding latest versions of the base image see
+# https://github.com/SwissDataScienceCenter/renkulab-docker
+ARG RENKU_BASE_IMAGE=renku/renkulab-vnc:0.10.3
+FROM ${RENKU_BASE_IMAGE}
+
+USER root
+
+# Install qGIS
```
Comment (rokroskar) [inline qgis/Dockerfile:8]: again, should maybe be a base image?

File: qgis/README.md [hunk 12]
```diff
@@ -0,0 +1,5 @@
+# QGIS Template
+
```
Comment (rokroskar) [inline qgis/README.md:2]: description block missing

File: qgis/README.md [hunk 13]
```diff
@@ -0,0 +1,5 @@
...
+## How to use this template
+
+This is a Renku project with QGIS pre-installed! You can access it through the `/vnc` end-point of your Renku session. You may be familiar with the `/lab` end-point, but you can simply access the vnc by changing the URL. Open up QGIS and away you go!
```
Comment (rokroskar) [inline qgis/README.md:5]: add an autolaunch link?

File: qgis/requirements.txt [hunk 14]
```diff
@@ -0,0 +1 @@
+PyQt5
```
Comment (rokroskar) [inline qgis/requirements.txt:1]: belongs in the base image

**Review by rokroskar — CHANGES_REQUESTED** [review]
These are great! We've been getting a lot of questions about the magic of VNC on renku... 

Mostly minor comments - the only big thing is whether or not we want to build base images separately. It bloats the base image repo/builds but otoh I don't the user project Dockerfiles should be installing functional things but rather only the user environment on top.

File: matlab/.gitignore (hunk 2 above)
Comment (gavin-k-lee) [reply matlab/.gitignore:338]: these won't be in the project directory afaik

**Comment by gavin-k-lee** [comment]
@rokroskar i've cleaned them up based on v0.11.0 of our Docker images. any other comments?

File: desktop/.dockerignore [hunk 15]
```diff
@@ -0,0 +1,4 @@
+data
+notebooks
+.renku
+.git
```
Comment (rokroskar) [inline desktop/.dockerignore:4]: still missing newlines... :)

File: desktop/Dockerfile [hunk 16]
```diff
@@ -0,0 +1,50 @@
+# For finding latest versions of the base image see
+# https://github.com/SwissDataScienceCenter/renkulab-docker
+ARG RENKU_BASE_IMAGE=renku/renkulab-vnc:0.10.3
```
Comment (rokroskar) [inline desktop/Dockerfile:3]:
```suggestion
ARG RENKU_BASE_IMAGE=renku/renkulab-vnc:0.11.0
```

File: desktop/Dockerfile [hunk 17]
```diff
@@ -0,0 +1,50 @@
...
+COPY requirements.txt environment.yml /tmp/
+RUN conda env update -q -f /tmp/environment.yml && \
+    /opt/conda/bin/pip install -r /tmp/requirements.txt && \
+    conda clean -y --all && \
+    conda env export -n "root"
+
+# RENKU_VERSION determines the version of the renku CLI
+# that will be used in this image. To find the latest version,
+# visit https://pypi.org/project/renku/#history.
+ARG RENKU_VERSION={{ __renku_version__ | default("0.16.2") }}
```
Comment (rokroskar) [inline desktop/Dockerfile:30]:
```suggestion
ARG RENKU_VERSION={{ __renku_version__ | default("1.0.5") }}
```

File: matlab/.renku/renku.ini [hunk 18]
```diff
@@ -0,0 +1,5 @@
+[interactive]
+default_url = /vnc
+cpu_request = 1
+mem_request = 4G
+disk_request = 2G
```
Comment (rokroskar) [inline matlab/.renku/renku.ini:5]: these seem pretty specific - are you sure you want to set them as defaults?

File: matlab/Dockerfile [hunk 19]
```diff
@@ -0,0 +1,50 @@
...
+COPY requirements.txt environment.yml /tmp/
+RUN conda env update -q -f /tmp/environment.yml && \
+    /opt/conda/bin/pip install -r /tmp/requirements.txt && \
+    conda clean -y --all && \
+    conda env export -n "root"
+
+# RENKU_VERSION determines the version of the renku CLI
+# that will be used in this image. To find the latest version,
+# visit https://pypi.org/project/renku/#history.
+ARG RENKU_VERSION={{ __renku_version__ | default("0.16.2") }}
```
Comment (rokroskar) [inline matlab/Dockerfile:30]:
```suggestion
ARG RENKU_VERSION={{ __renku_version__ | default("1.0.5") }}
```

File: matlab/requirements.txt (hunk 5 above)
Comment (rokroskar) [reply matlab/requirements.txt:2]: are these still needed? Presumably they are part of the base image?

File: qgis/Dockerfile [hunk 20]
```diff
@@ -0,0 +1,50 @@
...
+COPY requirements.txt environment.yml /tmp/
+RUN conda env update -q -f /tmp/environment.yml && \
+    /opt/conda/bin/pip install -r /tmp/requirements.txt && \
+    conda clean -y --all && \
+    conda env export -n "root"
+
+# RENKU_VERSION determines the version of the renku CLI
+# that will be used in this image. To find the latest version,
+# visit https://pypi.org/project/renku/#history.
+ARG RENKU_VERSION={{ __renku_version__ | default("0.16.2") }}
```
Comment (rokroskar) [inline qgis/Dockerfile:30]:
```suggestion
ARG RENKU_VERSION={{ __renku_version__ | default("1.0.5") }}
```

**Review by rokroskar — CHANGES_REQUESTED** [review]
Just a few minor touch-ups - and newlines 🏅

**Comment by rokroskar** [comment]
We should merge this and tag a new release with `0.4.0` - I've already made the change in the renku repo to make that the default.

File: matlab/.renku/renku.ini (hunk 18 above)
Comment (gavin-k-lee) [reply matlab/.renku/renku.ini:5]: Yeah Matlab dies if you don't set these high enough, so I think it's best to avoid people accidentally going too low

**Comment by gavin-k-lee** [comment]
Thanks @rokroskar - I've addressed all those little touch-ups.

File: matlab/.renku/renku.ini (hunk 18 above)
Comment (rokroskar) [reply matlab/.renku/renku.ini:5]: cool, all good just wanted to make sure it was not a leftover

**Pull request merged**